force-app/main/default/classes/SOQL.cls

```diff
@@ -314,28 +434,51 @@ public with sharing class SOQL {
      */
     public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
         // Query the original Opportunity Line Items
+        List<OpportunityLineItem> oppLineItems = [
+            SELECT Id, OpportunityId, Product2Id, Quantity, UnitPrice, PricebookEntryId
+            FROM OpportunityLineItem
+            WHERE OpportunityId = :origOppId
+        ];
 
         // Create a list to hold the new line items
+        List<OpportunityLineItem> clonedLineItems = new List<OpportunityLineItem>();
 
         // Generate a random number between 1 and 3
+        Integer random1To3 = (Integer)Math.ceil(Math.random() * 3);
 
         // Loop through each cloned Opportunity Id
-
-            // Loop through each original line item
-
-                // Clone the original line item and add the random number to the quantity
+        for (Id cloneOppId : cloneOppIds) {
+            for (OpportunityLineItem origLineItem : oppLineItems) {
+                // Clone the original line item
+                OpportunityLineItem newLineItem = new OpportunityLineItem(
+                    OpportunityId = cloneOppId,
+                    PricebookEntryId = origLineItem.PricebookEntryId,
+                    Quantity = origLineItem.Quantity + random1To3,
+                    UnitPrice = origLineItem.UnitPrice
+                );
+                clonedLineItems.add(newLineItem);
+            }
+        }
 
         // Insert the new line items
+        if (!clonedLineItems.isEmpty()) {
+            insert clonedLineItems;
+        }
     }
+
     
     /**
      * Question 24
      * Retrieves the Opportunity Name, Account Name, and Owner Name for Opportunities that have a Stage of 'Closed Won'.
      * Example: getOpportunityDetailsForClosedWon()
      * @return A list of filtered Opportunity
      */
-    public static List<Opportunity> getOpportunityDetailsForClosedWon() {        
-        return null; // Replace null with the variable you used to store the result
+    public static List<Opportunity> getOpportunityDetailsForClosedWon() {    
+        List<Opportunity> oppDetailsClosedWon = [
+        SELECT Name, AccountId, Account.Name, OwnerId, (SELECT Contact.Id, Contact.Name FROM OpportunityContactRoles)
```

Comment:
I'm not sure that the sub query is necessary here. Where are you using the OpportunityContactRoles?